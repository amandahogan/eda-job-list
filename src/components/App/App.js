import React, { PropTypes } from 'react'

import './App.css'
import Board from '../../containers/Board'
import Navbar from '../../components/Navbar'
import Welcome from '../../components/Welcome'

class App extends React.Component {
  render () {
    const { login, logout, profile } = this.props
    return (
      <div className='App'>
        <Navbar login={login} logout={logout} profile={profile} />
        {profile ? <Board /> : <Welcome />}
      </div>
    )
  }
}

App.propTypes = {
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  profile: PropTypes.object
}

export default App
