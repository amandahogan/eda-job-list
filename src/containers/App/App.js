import { connect } from 'react-redux'

import { login, logout } from '../../actions/auth'
import App from '../../components/App'

export default connect(
  state => {
    return {
      profile: state.auth
    }
  },
  dispatch => {
    return {
      login: profile => dispatch(login(profile)),
      logout: () => dispatch(logout())
    }
  }
)(App)
