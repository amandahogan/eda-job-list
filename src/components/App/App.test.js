import { shallow } from 'enzyme'
import React from 'react'

import App from './App'

describe('<App />', () => {
  it('logout matches snapshot', () => {
    const profile = null
    const login = () => {}
    const logout = () => {}
    expect(
      shallow(<App login={login} logout={logout} profile={profile} />).getNode()
    ).toMatchSnapshot()
  })
})

describe('<App />', () => {
  it('login matches snapshot', () => {
    const profile = {}
    const login = () => {}
    const logout = () => {}
    expect(
      shallow(<App login={login} logout={logout} profile={profile} />).getNode()
    ).toMatchSnapshot()
  })
})
