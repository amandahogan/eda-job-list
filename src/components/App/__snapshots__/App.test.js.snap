exports[`<App /> login matches snapshot 1`] = `
<div
  className="App">
  <Navbar
    login={[Function]}
    logout={[Function]}
    profile={Object {}} />
  <Connect(Board) />
</div>
`;

exports[`<App /> logout matches snapshot 1`] = `
<div
  className="App">
  <Navbar
    login={[Function]}
    logout={[Function]}
    profile={null} />
  <Welcome />
</div>
`;
